src/workerd/io/actor-cache.c++

```diff
@@ -185,34 +185,52 @@ kj::Maybe<kj::Promise<void>> ActorCache::getBackpressure() {
   return nullptr;
 }
 
-void ActorCache::requireNotOom() {
-  KJ_IF_MAYBE(e, oomException) {
+void ActorCache::requireNotTerminal() {
+  KJ_IF_MAYBE(e, maybeTerminalException) {
+    if (!gate.isBroken()) {
+      // We've tried to use storage after shutdown, break the output gate via `flushImpl()` so that
+      // we don't let the worker return stale state. This isn't strictly necessary but it does
+      // mirror previous behavior wherein we would use disabled storage via `flushImpl()` and break
+      // the output gate.
+      KJ_DBG("Scheduling a new flush because we tried to use an op");
```

Comment:
KJ_DBG left in

Reply:
Man, I really need a lint check for this.